### PR TITLE
Fix Muse Glimmer final norm initialization

### PR DESCRIPTION
**Pull request by yikw1022**
The final output norm uses `gain_center=0.0` and inherited zero weight
initialization from `_gain_norm`, resulting in zero effective scale at
initialization. This zeros the final hidden states and logits, and produces
zero gradients for the decoder and lm_head on the first backward pass.

Initialize only the final norm weight to one while retaining
`gain_center=0.0`, so its initial effective scale is one. This keeps the
checkpoint parameterization unchanged and leaves the other gain-centered
norm initializations unchanged.

Validation:
- `git diff --check` passed.
- A standalone RMSNorm forward/backward check produced zero outputs and
  gradients before the change, and nonzero outputs and gradients after it.

### CODE REVIEW

**Comment by tianyu-l** [comment]
@claude review this PR

File: torchtitan/models/muse_glimmer/__init__.py [hunk 1]
```diff
@@ -407,8 +407,11 @@ def _muse_glimmer_config(
             output_multiplier=output_multiplier,
             output_soft_cap_temp=20.0,
         ),
-        # Final output norm is gain-centered on 0.0.
-        norm=_gain_norm(dim, _NORM_EPS, gain_center=0.0),
+        # Keep the checkpoint's zero gain center, but initialize to unit scale.
+        norm=dataclasses.replace(
```
Comment (tianyu-l) [inline torchtitan/models/muse_glimmer/__init__.py:411]: see if you like this style or the style claude suggests

**Review by tianyu-l — APPROVED** [review]
might need to update golden loss files, please see what tests say
